chains/btc/listener/deposit-handler.go

```diff
@@ -0,0 +1,54 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package listener
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ChainSafe/sygma-relayer/relayer/transfer"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
+)
+
+type BtcDepositHandler struct{}
+
+// NewBtcDepositHandler creates an instance of BtcDepositHandler that contains
+// handler functions for processing deposit events
+func NewBtcDepositHandler() *BtcDepositHandler {
+	return &BtcDepositHandler{}
+}
+
+func (e *BtcDepositHandler) HandleDeposit(sourceID uint8,
+	depositNonce uint64,
+	resourceID [32]byte,
+	amount *big.Int,
+	data string,
+	blockNumber *big.Int,
+) (*message.Message, error) {
+
+	// data is composed of recieverEVMAddress_destinationDomainID
+	parsedData := strings.Split(data, "_")
+	evmAdd := common.HexToAddress(parsedData[0]).Bytes()
+	destDomainID, err := strconv.ParseUint(parsedData[1], 10, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	payload := []interface{}{
```

Comment:
Create a type for this payload that can be reused on destination. 
I think a TransferMessageData probably exist.

Reply:
I am using TransferMessageData, but payload is always just an interface, because the payload can be any data, depending on what kind of transfer we are doing. Erc20, GMP etc... then we parse the payload in the executor depending on the transfer type, right?

Reply:
Leave the type here as TransferMessageData because then we can just cast in the executor which is easier.
I think if you are using it as []interface here we cannot cast the type on the executor